Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -245,7 +245,8 @@ void MediaSource::completeSeek()
     // 4. Resume the seek algorithm at the "Await a stable state" step.
     m_private->seekCompleted();
 
-    monitorSourceBuffers();
+    if (m_pendingSeekTime.isInvalid())
+        monitorSourceBuffers();
```

Comment:
The call to `monitorSourceBuffers()` at the end of `completeSeek()` is intentional, as that's how the MediaSource readyState will be updated.

`monitorSourceBuffers()` is also called periodically by `HTMLMediaElement::playbackProgressTimerFired()`, but removing the call here may cause other problems.